dev-client/src/components/dialogs/ErrorDialog.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useMemo,
+  useState,
+} from 'react';
+import {useTranslation} from 'react-i18next';
+import {StyleSheet} from 'react-native';
+import {Modal, Portal} from 'react-native-paper';
+
+import DismissButton from 'terraso-mobile-client/components/buttons/DismissButton';
+import {Icon} from 'terraso-mobile-client/components/icons/Icon';
+import {ExternalLink} from 'terraso-mobile-client/components/links/ExternalLink';
+import {ModalHandle} from 'terraso-mobile-client/components/modals/Modal';
+import {Text, View} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {convertColorProp} from 'terraso-mobile-client/components/util/nativeBaseAdapters';
+
+export type ErrorDialogProps = React.PropsWithChildren<{
+  headline?: React.ReactNode;
```

Comment:
Is there a reason to use ReactNode here with TranslatedHeading as a prop to the ErrorDialog, as opposed to using a string here and TranslatedHeading as a component within the ErrorDialog?

Reply:
I thought it was easier to pass in a node rather than proliferate the "i18n key + optional bundle of values" parameters that the `typography` components currently take, which feels clunkier to me.